src/main/Tree.razor.cs

```diff
@@ -227,15 +225,21 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
                             }
                         }
 
-                        await Task.Run(() =>
+                        await Task.Run(async () =>
                         {
                             this.AvatarUrl = decodedUrl;
                             this.InitialRegionNeuron = regionNeuron;
                             this.InitialPostsynapticNeurons = postsynapticNeurons;
-                            this.Reload();
+                            await this.Reload();
                         });
 
                         Helper.ReinitializeOption(o => this.SelectedOption = o);
+
+                        if (query.TryGetValue("eupm", out var encodedEUPM) && this.Children.Any(x=>x.Neuron.Id == encodedEUPM))
```

Comment:
Please add "eupm" as a string constant inside a "Constants" struct in Types file.